Sources/SourceKitBazelBSP/SharedUtils/Shell/CommandRunner.swift

```diff
@@ -70,3 +70,30 @@ extension CommandRunner {
         return try bazel(baseConfig: baseConfig, rootUri: rootUri, cmd: cmd)
     }
 }
+
+// Define a protocol that both String and Data types can conform to
+protocol ContentRepresentable {
+    var asData: Data { get }
+    var asString: String { get }
+}
```

Comment:
This is how you can solve this problem without having each type declare its output:

```
protocol DataConvertible {
    func convert(data: Data) -> Self
}
```

```
extension Data: DataConvertible {
    func convert(data: Data) -> Self { self }
}

extension String: DataConvertible {
    func convert(data: Data) -> Self { (the string conversion) }
}
```

Then, you can use this type as a generic param:

```
public protocol CommandRunner {
    func run<T: DataConvertible>(_ cmd: String, cwd: String?) throws -> DataConvertible
}
```

That way you don't need the toStrings/toData :)